### DISCUSSION TITLE
Add a podman guide for using quadlet files

### DISCUSSION THREAD

**Discussion by madscientist16** [gotson/komga#1749] [Ideas]
I think it would be nice to have a guide with example [quadlet files](https://docs.podman.io/en/latest/markdown/podman-systemd.unit.5.html) for people using podman.

Here's some examples based on the quadlet files I use to run komga and komf.
### komga.pod
```
# Run komga and komf in a pod to make networking easier when running as rootless. 
# Komf can communicate with komga using localhost:25600 when both are running inside the same pod.

[Pod]
# Komf
PublishPort=8085:8085
# Komga
PublishPort=25600:25600
```
### komga.container
```
[Unit]
Description=Komga - Media server for comics & mangas

[Container]
## General
# AutoUpdate=registry allows this container to be updated using podman auto-update command
AutoUpdate=registry
Image=docker.io/gotson/komga:latest
Pod=komga.pod

## Volumes
# %h is a shortcut for the home directory 
Volume=%h/Server/komga-pod/komga:/config
Volume=%h/Documents/Books:/data

## Environment Variables
Environment="TZ=Your/Timezone"
Environment="JAVA_TOOL_OPTIONS=-Xmx4g"

[Service]
SuccessExitStatus=0 143
```
See [this post](https://serverfault.com/questions/695849/services-remain-in-failed-state-after-stopped-with-systemctl/695863#695863) for the reason for adding 143 as a success exit code
### komf.container
```
[Unit]
Description=Komf - Komga & Kavita Metadata Fetcher
After=komga.service

[Container]
# General
AutoUpdate=registry
Image=docker.io/sndxr/komf:latest
Pod=komga.pod

# Volumes
Volume=%h/Server/komga-pod/komf:/config

# Environment Variables
Environment="TZ=Your/Timezone"
Environment="KOMF_KOMGA_BASE_URI=http://localhost:25600"
Environment="KOMF_KOMGA_USER=admin@kethi.xyz"
Environment="JAVA_TOOL_OPTIONS=-XX:+UnlockExperimentalVMOptions -XX:+UseShenandoahGC -XX:ShenandoahGCHeuristics=compact -XX:ShenandoahGuaranteedGCInterval=3600000 -XX:TrimNativeHeapInterval=3600000"
# See link below on how to create podman secrets.
Secret=komga_password,type=env,target=KOMF_KOMGA_PASSWORD

[Service]
# Add 143 for the same reason as komga
SuccessExitStatus=0 143
```
See [this page](https://docs.podman.io/en/latest/markdown/podman-secret-create.1.html) for how to create podman secrets.

I have all three files above in `~/.config/containers/systemd/komga` directory. Then I run `systemctl --user daemon-reload` to generate the systemd services from the quadlet files. After that I just start the `komga-pod.service` with `systemctl --user start komga-pod.service`. If I wanted to start the komga pod when I boot up the server I could add the following to `komga.pod`
```
[Install]
WantedBy=default.target
```

**Comment by gotson** [comment]
Better open at https://github.com/gotson/komga-website

**Reply by madscientist16** [reply]
It says "please open issues on the main repo" on the readme file for komga-website that's why I didn't open an issue there.

**Reply by gotson** [reply]
> It says "please open issues on the main repo" on the readme file for komga-website that's why I didn't open an issue there.

that's not valid anymore, will change it
